### PR TITLE
golangci: add depguard

### PR DESCRIPTION
**Pull request by zirain**
Signed-off-by: hejianpeng <hejianpeng2@huawei.com>

### CODE REVIEW

File: tools/linter/golangci-lint/.golangci.yml [hunk 1]
```diff
@@ -4,18 +4,24 @@ run:
...
     - misspell
-    - exportloopref
+    - revive
     - unconvert
     - unparam
-    - goheader
-    - gocritic
 
 linters-settings:
+  depguard:
```
Comment (arkodg) [inline tools/linter/golangci-lint/.golangci.yml:20]: is this needed, I dont think we use `gogo` in this project ?
and reg `gopkg.in/yaml.v3` - we actually need to use this package in https://github.com/envoyproxy/gateway/pull/893

Comment (zirain) [reply tools/linter/golangci-lint/.golangci.yml:20]: prevent use `gogo` in the future.

Comment (zirain) [reply tools/linter/golangci-lint/.golangci.yml:20]: updated with `nolint` for special usecase.

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Pull request merged**